### PR TITLE
mutli: add release notes file to track in-repo

### PR DESCRIPTION
**Pull request by carlaKC**
This PR adds a file to track release notes in-repo. 
Open to suggestions to how we do this, but I think an append-only log will work?

For our next client release we will have a list like this:
```
# Next Release:
Feature 1
Feature 2
```

On release, we just shift the heading up:
```
# Next Release:

# v0.9.0
Feature 1
Feature 2
```

And just keep doing that. Decided on "Next Release" because we don't always know whether a change will go into a point release or a major one.

### CODE REVIEW

**Comment by carlaKC** [comment]
Requesting review from everybody, since the process affects us all. Feel free to remove yourself!

File: release_notes.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Loop Client Release Notes
+This file tracks release notes for the loop client. 
+
+Developers: 
+* When new features are added to the repo, a short description of the feature should be added under the Next Release heading.
```
Comment (guggero) [inline release_notes.md:5]: nit: put "Next Release" in quotes?

**Review by guggero — APPROVED** [review]
I like the proposed process! We should definitely try this and possibly also adopt in our other projects :100:

File: release_notes.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Loop Client Release Notes
+This file tracks release notes for the loop client. 
+
+Developers:
```
Comment (bhandras) [inline release_notes.md:4]: Consider making **Developers** and **Release Manager** bold to stand out better.

**Review by bhandras — APPROVED** [review]
Great idea!

**Comment by joostjager** [comment]
You could also add a PR template to the repo to remind us about adding the release notes.

**Comment by carlaKC** [comment]
Added a PR template (just with a reminder about notes, I don't think we need the full list that lnd has) and headings to the release notes (New Features/ Breaking Changes/ Bugfixes).

**Pull request merged**